### ISSUE TITLE
Virtual site interactions

### ISSUE THREAD

**Issue by lslumass** [openmm/openmm#4736]
Hi, 
I'm trying to set a series of virtual sites that belong to four different types A, B, C, D, and then define vdW or electrostatic interactions between each pair of them. I went through the documents about virtual sites in OpenMM, but still feel confused how to do. Is there any example I can follow? Thank you.

**Comment by peastman** [comment]
Can you describe in more detail what you want to do?  I'm not sure what you're asking for an example of.  If you mean a force field that uses virtual sites, look at one of the [four site water models](https://github.com/openmm/openmm/blob/master/wrappers/python/openmm/app/data/tip4pfb.xml), or for something much more complex the [CHARMM polarizable force field](https://github.com/openmm/openmm/blob/master/wrappers/python/openmm/app/data/charmm_polar_2019.xml).  If you're asking how to write code that creates virtual sites, that's described in [the API docs](http://docs.openmm.org/latest/api-python/).  Interactions for virtual sites are defined exactly the same way as for any other particles.

**Comment by lslumass** [comment]
Thank you for your response. In detail, I have three different planar molecules and am trying to add a virtual site for each molecule through the code to roughly mimic the pi-pi interaction among these molecules. In the API doc, I found setVirtualSite. Can we also set particle type, charge for these virtual site, just like add a particle or atom and set their properties? Thank you.

On Nov 23, 2024, at 11:01, Peter Eastman ***@***.***> wrote:

﻿
You don't often get email from ***@***.*** Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

Can you describe in more detail what you want to do? I'm not sure what you're asking for an example of. If you mean a force field that uses virtual sites, look at one of the four site water models<https://github.com/openmm/openmm/blob/master/wrappers/python/openmm/app/data/tip4pfb.xml>, or for something much more complex the CHARMM polarizable force field<https://github.com/openmm/openmm/blob/master/wrappers/python/openmm/app/data/charmm_polar_2019.xml>. If you're asking how to write code that creates virtual sites, that's described in the API docs<http://docs.openmm.org/latest/api-python/>. Interactions for virtual sites are defined exactly the same way as for any other particles.

**Comment by peastman** [comment]
How do you create your system?  Is it done entirely with code?  Or from a ForceField?  Or created by a different class such as AmberPrmtopFile?

What Forces does the System contain?

I suggest looking at the [manual](http://docs.openmm.org/latest/userguide/), especially chapter 8 which gives a brief overview of the architecture.  That will help you to understand what's going on.  There is no "particle type".  Properties like charge aren't actually properties of the particle, they're parameters of a particular Force object (most likely a NonbondedForce in that case) that computes forces based on them.

**Comment by lslumass** [comment]
Many thanks for your information. I might get the key point. I created the system through CHARMM pdb, psf, top, and parameter files of my own force field. Based on my understand, I defined an additional atom in the pdb file for each molecule, and then converted it into virtual site through code using ```system.setVirtualSite(idx, LocalCoordinatesSite(particles, (0.25, 0.25, 0.25, 0.25), (0,0,0,0), (0,0,0,0), (0,0,0)))``` . And the interactions between VSs were defined in the parameter files. It worked now. Thank you again and I will close the issue.
